File: src/task14_3_2/handlers/BaseHandler.java
package task14_3_2.handlers;

public class BaseHandler implements Handler{

    private Handler next;

    @Override
    public void setNext(Handler next) {
        this.next = next;
    }

    @Override
    public void handle(String filePath) {
        if (next != null){
            next.handle(filePath);
        }else{
            System.out.println("This format is unknown!!!");
        }
    }
}

File: src/task14_3_3/handlers/BaseHandler.kt
package task14_3_3.handlers

open class BaseHandler: Handler {

    private var next: Handler? = null

    override fun setNext(next: Handler): Handler {
        this.next = next
        return next
    }

    override fun handle(num1: Int, num2: Int, operation: Char): Int? {
        next?.let {
            return it.handle(num1, num2, operation)
        }
        return null
    }
}